sdk/core/core-amqp/src/requestResponseLink.ts

```diff
@@ -249,17 +231,17 @@ export class RequestResponseLink implements ReqResLink {
         );
         this.sender.send(request);
       });
-    const config: RetryConfig<AmqpMessage> = {
-      operation: sendRequestPromise,
-      connectionId: this.connection.id,
-      operationType:
-        request.to && request.to === Constants.cbsEndpoint
-          ? RetryOperationType.cbsAuth
-          : RetryOperationType.management,
-      delayInSeconds: options.delayInSeconds,
-      maxRetries: options.maxRetries
-    };
-    return retry<AmqpMessage>(config);
+
+    let result: AmqpMessage;
+    sendRequestPromise()
+      .then((message) => {
+        result = message;
+      })
+      .catch((err: Error) => {
+        throw err;
+      });
+
+    return result;
```

Comment:
This won't work. `sendRequestPromise` is async, so you'll return the `result` (which will be undefined) before you've set it. Why not just `return sendRequestPromise()`?

Reply:
I believe this is also one of the reasons for the build failure. 
@ramya0820 Did the error not show up locally when building? If so, then we should look into why that is so.

![image](https://user-images.githubusercontent.com/16890566/61341375-d17fff80-a7fa-11e9-9023-5d46ca7158b1.png)

Reply:
Agree, result needed to be set - fixing this

On a side note, I was confused if `return sendRequestPromise;` is right vs `return sendRequestPromise();` vs `return await sendRequestPromise();` in order to return a 'Promise' :-| 

I believe this here needs to be `return sendRequest;` and not `return sendRequest()` ? Because we invoke the promise and get reference to the message (and so to it's body) at outer statement `return retry<Message>(config).body` ? (At least, that is the objective, in order to remain consistent with what API was already defined as)

Reply:
The current function signature says that it should return a promise that is resolved with a message. So, you would need to make sure that's what the return value is.

`sendRequestPromise` is a function that when invoked, returns a promise that resolves with a message. If you return the result of invoking this function, you should get the expected return type.

Reply:
On an aside about the build, I've been pushing commits to test the build on PR .. there's been some problem with core-amqp building on my Windows PC and it's cumbersome to keep switching between Mac and PC workspaces. (Working on fixing this soon)